### PR TITLE
Decompile unk_0202D938

### PR DESCRIPTION
**Pull request by SinaMegapolis**
Pretty generic overlay related to Save_Array.c 
Was good decomp practice at least

### CODE REVIEW

File: src/unk_0202D938.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include "save.h"
+
+u32 sub_0202D938(void) {
+    return 0xe;
+}
+
+void sub_0202D93C(u8* argP) {
```
Comment (tgsm) [inline src/unk_0202D938.c:7]: This function takes in a struct*, not a u8*. This struct should be placed in a header file, along with declarations for each of the three functions here.

Something like this should be fine for now:
```c
typedef struct UnkSavStruct20 {
    u8 filler_00[14];
} UnkSavStruct20;
```

File: src/unk_0202D938.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include "save.h"
+
+u32 sub_0202D938(void) {
+    return 0xe;
```
Comment (tgsm) [inline src/unk_0202D938.c:4]:
```suggestion
    return sizeof(UnkSavStruct20);
```

File: src/unk_0202D938.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    argP[7] = 0;
+    argP[8] = 0;
+    argP[9] = 0;
+    argP[10] = 0;
+    argP[11] = 0;
+    argP[12] = 0;
+    argP[13] = 0;
+}
+
+void* sub_0202D95C(SAVEDATA* data) {
```
Comment (tgsm) [inline src/unk_0202D938.c:24]:
```suggestion
UnkSavStruct20* sub_0202D95C(SAVEDATA* data) {
```

File: src/unk_0202D938.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    return SavArray_get(data, 0x14);
```
Comment (tgsm) [inline src/unk_0202D938.c:25]:
```suggestion
    return SavArray_get(data, SAVE_UNK_20);
```

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/unk_0202D938.c (hunk 1 above)
Comment (SinaMegapolis) [reply src/unk_0202D938.c:7]: How can i tell when a function is taking in a struct in the future?

**Comment by SinaMegapolis** [comment]
made a header file, including the typedef struct
wasn't sure what SAVE_UNK_20 was meant to be so i defined it as a constant macro in the c file

**Comment by SinaMegapolis** [comment]
> made a header file, including the typedef struct wasn't sure what SAVE_UNK_20 was meant to be so i defined it as a constant macro in the c file

oh nvm i somehow missed the constants folder
removed the SAV_UNK_20 duplication

File: src/unk_0202D938.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#include "unk_0202D938.h"
+#include "save.h"
+#include "constants/save_arrays.h"
+
+u32 sub_0202D938(void) {
+    return sizeof(UnkSavStruct20);
+}
+
+void sub_0202D93C(UnkSavStruct20* ptr) {
+    UnkSavStruct20 unk_struct = *ptr;
```
Comment (tgsm) [inline src/unk_0202D938.c:10]: This line is probably redundant

File: src/unk_0202D938.c [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#include "unk_0202D938.h"
+#include "save.h"
+#include "constants/save_arrays.h"
+
+u32 sub_0202D938(void) {
+    return sizeof(UnkSavStruct20);
+}
+
+void sub_0202D93C(UnkSavStruct20* ptr) {
+    (*ptr).filler_00[0] = 0;
```
Comment (tgsm) [inline src/unk_0202D938.c:10]:
```suggestion
    ptr->filler_00[0] = 0;
```

File: src/unk_0202D938.c [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    ptr->filler_00[4] = 0;
+    ptr->filler_00[5] = 0;
+    ptr->filler_00[6] = 0;
+    ptr->filler_00[7] = 0;
+    ptr->filler_00[8] = 0;
+    ptr->filler_00[9] = 0;
+    ptr->filler_00[10] = 0;
+    ptr->filler_00[11] = 0;
+    ptr->filler_00[12] = 0;
+    ptr->filler_00[13] = 0;
```
Comment (PikalaxALT) [inline src/unk_0202D938.c:23]: This looks like an expanded `memset`.

```suggestion
    memset(ptr, 0, sizeof(UnkSavStruct20));
```

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/unk_0202D938.c (hunk 7 above)
Comment (SinaMegapolis) [reply src/unk_0202D938.c:23]: oh, TIL that function exists

**Review by PikalaxALT — APPROVED** [review]

**Review by tgsm — APPROVED** [review]

**Pull request merged**